### PR TITLE
Link to release docs

### PR DESCRIPTION
**Pull request by 2m**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ This project provides a home to Akka Streams connectors to various technologies,
 Documentation
 -------------
 
-Latest [documentation](http://developer.lightbend.com/docs/alpakka/latest/).
+See [Releases](https://github.com/akka/alpakka/releases) page for the reference and API documentation.
```
Comment (patriknw) [inline README.md]: Hm, I'm not sure. A direct link to latest released documentation is much better. I didn't know that `latest` was master. If there is no support for such `current` symlink in techhub we should open an issue, and until solved we need to update this link manually.

That said, we can add more links here, one to the the Releases and another to the API docs.

Comment (johanandren) [inline README.md]: Oh, I'm to tired to review stuff. I my brain auto-corrected this to a different release path on techhub... I agree with Patrik.

Comment (2m) [inline README.md]: I like this, because it allows us to have one less step when cutting a release. And releases page has some useful information.

Comment (johanandren) [inline README.md]: I wish there was a `current` in techhub. Could we open a ticket for that?

Comment (patriknw) [inline README.md]: Until there is a `current` we must think about users, and not only what is least work for us. This is currently the landing page and there should be quick links to the documentation.

Comment (2m) [inline README.md]: Fair enough. Will change this to point to the current version.

**Pull request merged**